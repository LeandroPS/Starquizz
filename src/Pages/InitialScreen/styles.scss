
.initial-screen{
    text-align: center;
    background-color: #000;
    color: #fff;

    width: 100%;
    height: 100vh;

    .darth-head{
        width: 300px;
    }

    h1{
        font-family: 'Star Wars Font', sans-serif;
    }

    h2{
        font-weight: lighter;

    }
}
