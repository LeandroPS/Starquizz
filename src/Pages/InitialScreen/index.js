import React, {Component} from 'react'
import PropTypes from 'prop-types';

import '../../styles/common.scss';
import './styles.scss';

class Initialscreen extends Component{

    render(){
        const {onStart} = this.props;

        return(
            <div className="initial-screen">
                <div className="centered">
                    <img 
                        className="darth-head"
                        src={require('../../assets/darth.png')}
                        alt="A cabeça do Darth Vader"/>
                    <h1>Star Wars quizz</h1>
                    <h2>Com esse quiz você terá oportunidade de identificar os principais personagens de Star- wars, marcar pontos e se tornar um expert nesta série de filmes maravilhosa!</h2>
                    <button 
                        className="button start-button" 
                        onClick={()=>onStart()}>
                        START
                    </button>
                </div>
            </div>
        );
    }
}

Initialscreen.propTypes = {
    onStart: PropTypes.func.isRequired
}

export default Initialscreen;