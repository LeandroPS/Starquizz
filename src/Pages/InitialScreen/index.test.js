import React from 'react';
import ReactDOM from 'react-dom';
import { create } from "react-test-renderer";
import { shallow } from 'enzyme';
import InitialScreen from './';

describe('Initial screen', () => {
    const props = {
        onStart: jest.fn()
    }

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<InitialScreen {...props}/>, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders correctly', () => {
        const tree = create(<InitialScreen {...props}/>)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('executes onStart function when clicking on start button', () => {
        const component = shallow(<InitialScreen {...props}/>);

        const button = component.find("button");
        button.simulate("click");

        expect(props.onStart).toHaveBeenCalled();

    });
});