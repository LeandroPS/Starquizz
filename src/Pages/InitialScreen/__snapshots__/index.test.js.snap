// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Initial screen renders correctly 1`] = `
<div
  className="initial-screen"
>
  <div
    className="centered"
  >
    <img
      alt="A cabeça do Darth Vader"
      className="darth-head"
      src="darth.png"
    />
    <h1>
      Star Wars quizz
    </h1>
    <h2>
      Com esse quiz você terá oportunidade de identificar os principais personagens de Star- wars, marcar pontos e se tornar um expert nesta série de filmes maravilhosa!
    </h2>
    <button
      className="button start-button"
      onClick={[Function]}
    >
      START
    </button>
  </div>
</div>
`;
